app/models/match.rb

```diff
@@ -93,46 +80,52 @@ def add_players_for_match(red_att, red_def, blue_att, blue_def)
     self.blue_def = blue_def
   end
 
-  # helper
-  def self.get_json_from_matches_array(matches)
-    match_json_arr = []
+  # helpers
+  def self.get_matches_arr(matches)
+    # @return matches array
+    match_info_arr = []
     matches.each do |match|
-      match_json_arr.push(get_json_from_match(match))
+      match_info_arr.push(get_match_hash(match))
     end
-    return match_json_arr
+    return match_info_arr
   end
 
-  def self.get_json_from_match(match)
-    match_pm_records = PlayerMatch.where(:match_id => match.id)
-    # puts match_pm_records
-
-    # Setup json and append
-    red_atk = Player.find((match_pm_records.where(:position => 'atk', :team => 'r').take!).player_id)
-    red_def = Player.find((match_pm_records.where(:position => 'def', :team => 'r').take!).player_id)
-    blue_atk = Player.find((match_pm_records.where(:position => 'atk', :team => 'b').take!).player_id)
-    blue_def = Player.find((match_pm_records.where(:position=> 'def', :team => 'b').take!).player_id)
-
-    # red_atk_alias = Player.find((match_pm_records.where(:position => 'atk', :team => 'r').take!).player_id).alias
-    # blue_atk_alias = Player.find((match_pm_records.where(:position => 'atk', :team => 'b').take!).player_id).alias
-    # red_def_alias = Player.find((match_pm_records.where(:position => 'def', :team => 'r').take!).player_id).alias
-    # blue_def_alias = Player.find((match_pm_records.where(:position => 'def', :team => 'b').take!).player_id).alias
-
-    json_hash = {
-        :match_id => match.id,
-        :red_atk => red_atk.alias,
-        :red_def => red_def.alias,
-        :blue_atk => blue_atk.alias,
-        :blue_def => blue_def.alias,
-        # :red_atk_alias => red_atk_alias,
-        # :blue_atk_alias => blue_atk_alias,
-        # :red_def_alias => red_def_alias,
-        # :blue_def_alias => blue_def_alias,
-        :red_goal => match.redGoal,
-        :blue_goal => match.blueGoal,
-        :winner => match.winner,
-        :date => match.created_at.strftime('%d %^b %Y')
-    }
+  def self.get_player_from_pm(players, pm)
+    players.each do |player|
+      if pm.player_id == player.id
+        return player
+      end
+    end
+  end
 
-    return json_hash
+  def self.get_match_hash(match)
+    # @return single match hash
+    player_matches = match.player_matches
+    players = match.players
+    red_atk ||= []
+    red_def ||= []
+    blue_atk ||= []
+    blue_def ||= []
+
+    # find player with associated team and position
+    player_matches.each do |pm|
+      if pm.position == 'atk' and pm.team == 'r'
```

Comment:
use `&&` instead of `and`. It has different meaning actually. Use `and` and `or` for flow control (eg. `do_something and return`) and `&&` for logical operators.
http://rubyinrails.com/2014/01/30/difference-between-and-and-in-ruby/